### ISSUE TITLE
[a11y] Charts: Filters in cards are not announced by screen readers

### ISSUE THREAD

**Issue by rak-phillip** [rancher/dashboard#16581]
**Setup**
- Rancher version: v2.14-head
- Rancher UI Extensions: NA
- Browser type & version: Firefox/Chrome

**Describe the bug**

Each card is accessible by keyboard; the card contents are also accessible by keyboard. When a tag like "Partners" or "Linux Only" receives focus, the screen reader announces the parent card action.

**To Reproduce**

1. Navigate to Local Cluster => Apps => Charts
2. Navigate to a tag within a card, like "Linux Only", using the keyboard while screen reader is active

**Result**

The screen reader will announce the contents of the parent card.

**Expected Result**

The screen reader should announce the contents of the tag that has focus, for example: "Linux Only"

**Screenshots**

NA

**Additional context**

A workaround exists: Filters can be applied and announced via the sidebar to the right.

**Comment by rak-phillip** [comment]
@gaktive @nwmac this could be a good backport candidate for v2.13.3

**Comment by a-arias** [comment]
**To Test**
Fix validated ✅
Rancher : v2.14-c533e5bb4c0c346f92db7096882397e8e2548e99-head
UI: master 388d323

https://github.com/user-attachments/assets/7f8d9eb5-2fdf-42d3-a111-ef440ce9a3e9
